src/PactNet.AspNetCore/Messaging/MessageMiddleware.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using PactNet.AspNetCore.Messaging.Models;
+using PactNet.AspNetCore.Messaging.Options;
+using PactNet.Native;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace PactNet.AspNetCore.Messaging
+{
+    /// <summary>
+    /// Defines the message middleware
+    /// It handles the queries for the verifier when there are message interactions
+    /// </summary>
+    public class MessageMiddleware
+    {
+        /// <summary>
+        /// The default serializer options
+        /// </summary>
+        public static JsonSerializerOptions SerializerOptions => new JsonSerializerOptions
```

Comment:
This is using `System.Text.Json` but I'm pretty sure the main `PactNet` repo uses Newtonsoft doesn't it? We should probably make sure we're only using one.

Reply:
lol you literally told me you'd prefer to use system.text in the other PR :D 
Let's make up our mind once for all about which lib we want to use. I'm all for newtonsoft though!

Reply:
Newtonsoft approach pushed